File: android/hovr_device_permissions/src/main/kotlin/com/hovr/devicepermissions/PermissionStatus.kt
package com.hovr.devicepermissions

enum class PermissionStatus {
    GRANTED,
    DENIED,
    DENIED_PERMANENTLY,
    NOT_DETERMINED,
    RESTRICTED,
    SERVICE_DISABLED,
}

enum class AlertPriority(val level: Int) {
    NETWORK(0),
    LOCATION(1),
    NOTIFICATION(2),
}

File: ios/Sources/HovrDevicePermissions/PermissionStatus.swift
import Foundation

public enum PermissionStatus: Equatable {
    case granted
    case denied
    case deniedPermanently
    case notDetermined
    case restricted
    case serviceDisabled
}

public enum AlertPriority: Int {
    case network = 0
    case location = 1
    case notification = 2
}

public enum PermissionLimits {
    public static let notificationPromptDelayNanoseconds: UInt64 = 400_000_000
}
